editor/editor_help_search.cpp

```diff
@@ -590,27 +657,54 @@ TreeItem *EditorHelpSearch::Runner::_create_class_hierarchy(const ClassMatch &p_
 	return class_item;
 }
 
+bool EditorHelpSearch::Runner::_find_or_create_item(TreeItem *p_parent, const String &p_item_meta, TreeItem *&r_item) {
+	// Attempt to find in cache.
+	if (tree_cache->item_cache.has(p_item_meta)) {
+		r_item = tree_cache->item_cache[p_item_meta];
+
+		// Remove from cache.
+		tree_cache->item_cache.erase(p_item_meta);
+
+		// Add to tree.
+		p_parent->add_child(r_item);
+
+		return false;
+	} else {
+		// Otherwise create item.
+		r_item = results_tree->create_item(p_parent);
+
+		return true;
+	}
+}
+
 TreeItem *EditorHelpSearch::Runner::_create_class_item(TreeItem *p_parent, const DocData::ClassDoc *p_doc, bool p_gray) {
 	String tooltip = DTR(p_doc->brief_description.strip_edges());
 
-	TreeItem *item = results_tree->create_item(p_parent);
-	item->set_icon(0, EditorNode::get_singleton()->get_class_icon(p_doc->name));
-	item->set_text(0, p_doc->name);
-	item->set_text(1, TTR("Class"));
-	item->set_tooltip_text(0, tooltip);
-	item->set_tooltip_text(1, tooltip);
-	item->set_metadata(0, "class_name:" + p_doc->name);
+	const String item_meta = "class_name:" + p_doc->name;
+
+	TreeItem *item = nullptr;
+	if (_find_or_create_item(p_parent, item_meta, item)) {
+		item->set_icon(0, EditorNode::get_singleton()->get_class_icon(p_doc->name));
+		item->set_text(0, p_doc->name);
+		item->set_text(1, TTR("Class"));
+		item->set_tooltip_text(0, tooltip);
+		item->set_tooltip_text(1, tooltip);
+		item->set_metadata(0, item_meta);
+		if (p_doc->is_deprecated) {
+			Ref<Texture2D> error_icon = ui_service->get_editor_theme_icon("StatusError");
```

Comment:
Will add the `SNAME` change here instead as this is the area where the code is changed, also in a future push